catalyst_voices/lib/widgets/cards/role_chooser_card.dart

```diff
@@ -0,0 +1,241 @@
+  final bool lockValueAsDefault;
+
+  /// Hides the "Learn More" link.
+  final bool isLearnMoreHidden;
+
+  /// Toggles view-only mode.
+  final bool isViewOnly;
+
+  /// A callback triggered when the role selection changes.
+  final ValueChanged<bool>? onChanged;
+
+  /// A callback triggered when the "Learn More" link is clicked.
+  final VoidCallback? onLearnMore;
+
+  const RoleChooserCard({
+    super.key,
+    required this.value,
+    required this.imageUrl,
+    required this.label,
+    this.lockValueAsDefault = false,
+    this.isLearnMoreHidden = false,
+    this.isViewOnly = false,
+    this.onChanged,
+    this.onLearnMore,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: 100,
+      decoration: isViewOnly
+          ? null
+          : BoxDecoration(
+              border: Border.all(
+                color: Theme.of(context).colors.outlineBorderVariant!,
+                width: 1,
+              ),
+              borderRadius: BorderRadius.circular(8),
+            ),
+      child: Padding(
+        padding: EdgeInsets.all(isViewOnly ? 8 : 12),
+        child: Row(
+          children: [
+            Column(
+              children: [
+                ColorFiltered(
+                  colorFilter: _grayscaleFilter,
+                  child: CatalystImage.asset(
+                    imageUrl,
+                    width: 70,
+                    height: 70,
+                    fit: BoxFit.cover,
+                  ),
+                ),
+              ],
+            ),
+            const SizedBox(width: 12),
+            Expanded(
+              child: Column(
+                children: [
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      Expanded(
+                        child: Text(
+                          overflow: TextOverflow.ellipsis,
+                          label,
+                          style: Theme.of(context).textTheme.titleSmall,
+                        ),
+                      ),
+                      if (!isLearnMoreHidden) ...[
+                        const SizedBox(width: 10),
+                        GestureDetector(
+                          onTap: onLearnMore,
+                          child: Text(
+                            context.l10n.learnMore,
+                            style: Theme.of(context)
+                                .textTheme
+                                .labelMedium!
+                                .copyWith(
+                                  color: Theme.of(context).colors.iconsPrimary,
+                                ),
+                          ),
+                        ),
+                      ],
+                    ],
+                  ),
+                  const SizedBox(height: 8),
+                  Row(
+                    children: [
```

Comment:
Those widgets are kinda hard to read. Try to split them into smaller, private, widgets.